### DISCUSSION TITLE
EB/CNS Test

### DISCUSSION THREAD

**Discussion by jramirezcr** [AMReX-Codes/amrex#2146] [Q&A]
Hello dear AMRex community.

I want to use AMR Framework in my code, first I am trying to understand the **EB/CNS** tutorial. So I would like to know if you have any reference about the numerical scheme used in this test.

Thank you very much
Regards
JRC

**Comment by asalmgren** [comment]
Jorge,

Welcome to the AMReX community!

Can you tell us a little bit about what type of problem you want to solve
with AMReX?  Will it be compressible flow?

An

On Sat, Jul 3, 2021 at 7:18 PM Jorge Ramírez Cruz ***@***.***>
wrote:

> Hello dear AMRex community.
>
> I want to use AMR Framework in my code, first I am trying to understand
> the *EB/CNS* tutorial. So I would like to know if you have any reference
> about the numerical scheme used in this test.
>
> Thank you very much
> Regards
> JRC
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/AMReX-Codes/amrex/discussions/2146>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACRE6YTCMYAYKUVRSVHWJF3TV7AITANCNFSM47YWRHQA>
> .
>


-- 
Ann Almgren
Senior Scientist; CCSE Group Lead
Pronouns: she/her/hers

**Reply by jramirezcr** [reply]
Hi An

It is a compressible flow (multiple jets injected into a mixing chamber). 

JRC

**Comment by asalmgren** [comment]
JRC,

I would suggest looking at the PeleC code (based on AMReX) for compressible
reacting flow with EB geometries -- I think they probably have some
problems with multiple jets even set up already ... no point reinventing
the wheel :-)

On Sat, Jul 3, 2021 at 7:28 PM Jorge Ramírez Cruz ***@***.***>
wrote:

> Hi An
>
> It is a compressible flow (multiple jets injected into a mixing chamber).
>
> JRC
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/AMReX-Codes/amrex/discussions/2146#discussioncomment-961738>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACRE6YXXMSUHIHY5UNE3BNDTV7BOFANCNFSM47YWRHQA>
> .
>


-- 
Ann Almgren
Senior Scientist; CCSE Group Lead
Pronouns: she/her/hers

**Reply by drummerdoc** [reply]
Happy to help. Let know if you run into problems. There’s even a paper that describes the code (although it didn’t make the cut at SC this year, unfortunately).

**Reply by jramirezcr** [reply]
Dear Ann and Marc.

The thing is that our team has been developing different solvers for years. However, they are curvilinear and use immersed boundaries to represent the geometries. Now, we are trying to port them using AMR techniques. Currently, we have three options: 

-Develop the entire code (not the best idea).
-Use PARAMESH
-Use the AMRex framework.

We decided to use your framework AMRex. The first solver we wanted to port is a fully explicit compressible code. That's why I ask for references on the EB/CNS test to use it as a guide (just the numerical scheme implemented). The documentation of the framework is pretty good but we are trying to understand in depth this case.

Thank you so much
Saludos Cordiales 
JRC

**Comment by asalmgren** [comment]
Understood!

Tests/EB/CNS is a compressible solver using method-of-lines with Riemann
solver (not sure which one).   Unfortunately it's not the best example of
how to code in AMReX since it still uses f90 code which we have tried to
get away from.   (We've been meaning to migrate it to our C++ style but
just haven't had time yet.)

It does have AMR with subcyling in time, EB and flux redistribution (and
re-redistribution at coarse-fine boundaries)

Hope that helps!

Ann

p.s.  If you'd like to be able to ask questions as you go along, we'd be
happy to invite you to our amrex slack channel...



On Sun, Jul 4, 2021 at 4:51 PM Jorge Ramírez Cruz ***@***.***>
wrote:

> Dear Ann and Marc.
>
> The thing is that our team has been developing different solvers for
> years. However, they are curvilinear and use immersed boundaries to
> represent the geometries. Now, we are trying to port them using AMR
> techniques. Currently, we have three options:
>
> -Develop the entire code (not the best idea).
> -Use PARAMESH
> -Use the AMRex framework.
>
> We decided to use your framework AMRex. The first solver we wanted to port
> is a fully explicit compressible code. That's why I ask for references on
> the EB/CNS test to use it as a guide (just the numerical scheme
> implemented). The documentation of the framework is pretty good but we are
> trying to understand in depth this case.
>
> Thank you so much
> Saludos Cordiales
> JRC
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/AMReX-Codes/amrex/discussions/2146#discussioncomment-963780>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACRE6YWTBUGUINA7W3HXNO3TWDXWLANCNFSM47YWRHQA>
> .
>


-- 
Ann Almgren
Senior Scientist; CCSE Group Lead
Pronouns: she/her/hers

**Reply by jramirezcr** [reply]
It helps to me, thanks. And it would be nice to be in the AMRerx slack channel :) .

**Comment by asalmgren** [comment]
Jorge - can you email me your preferred email for the slack invite? thanks

Ann

On Mon, Jul 5, 2021 at 8:39 PM Jorge Ramírez Cruz ***@***.***>
wrote:

> It helps to me, thanks. And it would be nice to be in the AMRerx slack
> channel :) .
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/AMReX-Codes/amrex/discussions/2146#discussioncomment-968220>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACRE6YS65KSGZ4TKMKC25G3TWJ3G3ANCNFSM47YWRHQA>
> .
>


-- 
Ann Almgren
Senior Scientist; CCSE Group Lead
Pronouns: she/her/hers

**Reply by jramirezcr** [reply]
Sure, jramirezcr@iingen.unam.mx

Thanks.
JRC
